xbmc/interfaces/AnnouncementManager.h

```diff
@@ -46,14 +46,14 @@ namespace ANNOUNCEMENT
     void RemoveAnnouncer(IAnnouncer *listener);
 
     void Announce(AnnouncementFlag flag, const char *sender, const char *message);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CVariant &data);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CFileItemPtr item);
-    void Announce(AnnouncementFlag flag, const char *sender, const char *message, CFileItemPtr item, CVariant &data);
+    void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CVariant &data);
+    void Announce(AnnouncementFlag flag, const char *sender, const char *message, const CFileItemPtr item);
```

Comment:
`item` is a value parameter so adding const in this context is pretty pointless.

I would check if it's possible to make the parameter `shared_ptr<const CFileItem>` instead, as it should have been from the beginning. If it requires too much changes, then fallback to copying.